transaction.go

```diff
@@ -283,6 +283,9 @@ func (t *Transaction) SignalUpdatesApplied() error {
 	}
 	t.wal.mu.Unlock()
 
+	// Decrease the number of active transactions
+	atomic.AddInt64(&t.wal.atomicUnfinishedTxns, -1)
```

Comment:
It's worth adding a check here to make sure it doesn't equal -1, panicking if it does.